hack/kcp/resource_quota_controller_patch.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"log"
+	"strings"
+)
+
+/*
+Process:
+
+1. go run ./hack/kcp/resource_quota_controller_patch.go > pkg/controller/resourcequota/resource_quota_controller_patch.go
+(you may need to add -mod=readonly)
+
+2. goimports -w pkg/controller/resourcequota/resource_quota_controller_patch.go
+
+3. reapply patch for kcp to pkg/controller/resourcequota/resource_quota_controller_patch.go
+*/
+
+func main() {
+	fileSet := token.NewFileSet()
+
+	file, err := parser.ParseFile(fileSet, "pkg/controller/resourcequota/resource_quota_controller.go", nil, parser.ParseComments)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// n stores a reference to the node for the function declaration for Sync
+	var n ast.Node
+
+	ast.Inspect(file, func(node ast.Node) bool {
+		switch x := node.(type) {
+		case *ast.FuncDecl:
+			if x.Name.Name == "Sync" {
+				// Rename the function
+				x.Name.Name = "KCPUpdateMonitors"
```

Comment:
is there really value in this rather than just a patch doing the same?

Reply:
I can do it in the patch